### PR TITLE
ENH: Enable model conversion from TreeLite

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR extends the model builders module in daal4py to be able to convert from objects in TreeLite format.

By supporting this format, models from other libraries such as scikit-learn can be converted to daal4py using this as an intermediate, without having to support those libraries directly.

It also allows supporting dart boosters from xgboost and lightgbm along the way, since treelite already takes care of the right conversion into a format that is very similar to daal4py's.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

Not applicable.

### CODE REVIEW

**Comment by razdoburdin** [comment]
Is it possible to replace original xgboost and lgbm convertors by the new one for treelite?

**Comment by david-cortes-intel** [comment]
> Is it possible to replace original xgboost and lgbm convertors by the new one for treelite?

Yes, but it would require adding treelite as a hard dependency. But why would you want to do that?

**Comment by razdoburdin** [comment]
> Yes, but it would require adding treelite as a hard dependency.
Got it.
>  But why would you want to do that?
I look for some ways to generalize conversion (not high priority).

**Review by razdoburdin — APPROVED** [review]
great job.
